### PR TITLE
Feature/dynamically loaded xr rigs

### PR DESCRIPTION
**Pull request by SimonTheSourcerer**
### Description
Allows setup of the rig dynamically.
Needs this PR to work: https://github.com/Innoactive/Basic-Interaction-Component/pull/37

### CODE REVIEW

**Review by Gusinuhe — CHANGES_REQUESTED** [review]

File: Editor/Rigs/XRSimulatorSetup.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+            GameObject deviceSimulator = GameObject.Instantiate(FindPrefab("[XR Device Simulator]"));
+            deviceSimulator.transform.position = Vector3.zero;
+        }
+
+        /// <inheritdoc />
+        public override bool CanBeUsed()
+        {
+#if ENABLE_INPUT_SYSTEM
+#pragma warning disable 618
+            return IsEventManagerInScene() == false && PlayerSettings.virtualRealitySupported == false;
```
[the remark this answers is not in the document]
Comment (SimonTheSourcerer) [reply Editor/Rigs/XRSimulatorSetup.cs:30]: I have not seen anything from the XR Plugin Management preventing me to start the simulator

File: Runtime/Rigs/XRSetupBase.cs [hunk 2]
```diff
@@ -0,0 +1,14 @@
+﻿using Innoactive.Creator.BasicInteraction.RigSetup;
+using UnityEngine;
+using UnityEngine.XR.Interaction.Toolkit;
+
+namespace Innoactive.Creator.Components.Runtime.Rigs
+{
+    public abstract class XRSetupBase : InteractionRigProvider
```
Comment (Gusinuhe) [inline Runtime/Rigs/XRSetupBase.cs:7]: Missing docs

**Comment by Gusinuhe** [comment]
Rigs are not instantiated and the following error is produced: 

image

File: Editor/XRInteractionSceneSetup.cs [hunk 3]
```diff
@@ -1,26 +1,30 @@
...
-            RemoveMainCamera();
-#if ENABLE_LEGACY_INPUT_MANAGER && XRIT_0_10_OR_NEWER
-            SetupPrefab("[XR_Setup_Action_Based]");
-#else
-            SetupPrefab("[XR_Setup_Device_Based]");
-#endif
+            GameObject oldRig = GameObject.Find("[XR_Setup]");
+            if (oldRig != null)
+            {
+                Debug.Log("TEST");
```
Comment (Gusinuhe) [inline Editor/XRInteractionSceneSetup.cs:22]: TEST!!

**Review by Gusinuhe — APPROVED** [review]

**Review by Gusinuhe — APPROVED** [review]

**Pull request merged**